src/main/java/randoop/contract/EqualsTransitive.java

```diff
@@ -0,0 +1,62 @@
+package randoop.contract;
+
+import randoop.Globals;
+
+/**
+ * The contract:
+ * <code>!(x0.equals(x1) &amp;&amp; x1.equals(x2))||x0.equals(x2)</code>.
```

Comment:
I would find the specification easier to read as an implication rather than when converted into more primitive operators (though the latter is how it would be implemented).

The spacing is confusing, with the || jammed against its operators even though it is the top-level operator.